bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BIntersectionType.java

```diff
@@ -55,19 +56,32 @@ public class BIntersectionType extends BType implements IntersectionType {
 
     public BIntersectionType(Module pkg, Type[] constituentTypes, Type effectiveType,
                              int typeFlags, boolean readonly) {
-        this(pkg, constituentTypes, typeFlags, readonly);
+        this(TypeConstants.INTERSECTION_TNAME, pkg, constituentTypes, typeFlags, readonly);
         this.effectiveType = effectiveType;
     }
 
     public BIntersectionType(Module pkg, Type[] constituentTypes, IntersectableReferenceType effectiveType,
                              int typeFlags, boolean readonly) {
-        this(pkg, constituentTypes, typeFlags, readonly);
+        this(TypeConstants.INTERSECTION_TNAME, pkg, constituentTypes, typeFlags, readonly);
         this.effectiveType = effectiveType;
         effectiveType.setIntersectionType(this);
     }
 
-    private BIntersectionType(Module pkg, Type[] constituentTypes, int typeFlags, boolean readonly) {
-        super(null, pkg, Object.class);
+    public BIntersectionType(String typeName, Module pkg, Type[] constituentTypes, Type effectiveType,
+                             int typeFlags, boolean readonly) {
+        this(typeName, pkg, constituentTypes, typeFlags, readonly);
+        this.effectiveType = effectiveType;
+    }
+
+    public BIntersectionType(String typeName, Module pkg, Type[] constituentTypes,
```

Comment:
Seems no usage of these new two constructors?

Reply:
Those are used by generated bytecode